syntax = "proto3";

package csce438;
import "google/protobuf/timestamp.proto";

service CoordService{
  rpc Heartbeat (ServerInfo) returns (Confirmation) {}
  rpc GetServer (ID) returns (ServerInfo) {}
  // ZooKeeper API here

  // Zookeeper-like API
  // Create a path and place data in the znode
  rpc Create (PathAndData) returns (Confirmation) {}

  // Check if a path exists (checking if a Master is elected)
  //rpc exists (Path) returns (Status) {}
}

//server info message definition
message ServerInfo{
  int32 serverID = 1;
  string hostname = 2;
  string port = 3;
  string type = 4;
}

//confirmation message definition
message Confirmation{
  bool status = 1;
}

//id message definition
message ID{
  int32 id = 1;
}

message PathAndData {
  string path = 1;
  ServerInfo data = 2;
}

//message Status {
//   string status = 1;
//}

//message Path {
//    string path = 1;
//}